src/utils/yara.js

```diff
@@ -0,0 +1,76 @@
+import yara from 'yara';
+
+export default class YaraManager {
+  constructor() {
+    this.resources = [];
+  }
+
+  /*
+   * This Method scans a list of raw js artifacts using
+   * a set of Yara rules. Returns the Js artifacts that match
+   * otherwise, no malware was detected.
+   */
+  setupResourceScan(jsArtifacts, ticket) {
+    console.log('Scanning JS...');
+    const ticketId = ticket.getID();
+
+    yara.initialize(function(error) {
+      if (error) {
+        console.log(error.message);
+      } else {
+        const rules = [{ filename: 'src/resources/rules.yara' }];
+
+        const scanner = yara.createScanner();
+        scanner.configure({ rules: rules }, function(error, warnings) {
+          if (error) {
+            if (error instanceof yara.CompileRulesError) {
+              console.log(error.message + ': ' + JSON.stringify(error.errors));
```

Comment:
We should return the error -  so it can be either returned as part of the response, or so the request fails out.

Reply:
If there's a more complex return result, we could also encode the errors as part of that.

Reply:
what do you think about about returning just a simple JSON `{"error":true}` as this way we know the hc has encountered an error in the frontend but we keep the error info hidden to only the system logs? None of these errors are really useful to the user, they are more developer oriented...

Reply:
@williamtdr

Reply:
Yeah, sounds good to me!